src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,27 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        listeners.forEach(e -> executor.execute(() -> e.onEvent(event)));
     }
 
     public void shutdown() {
-
+        executor.shutdown();
     }
```

Comment:
When shutting down the ExecutorService, it's a good practice to also wait for previously submitted tasks to complete and to handle InterruptedException. You might want to use 'awaitTermination' after 'shutdown' to ensure that all tasks have completed before the application exits.